packages/mui-material/src/version/index.ts

```diff
@@ -0,0 +1,8 @@
+export const version = process.env.MUI_VERSION;
+export const major = Number(process.env.MUI_MAJOR_VERSION);
+export const minor = Number(process.env.MUI_MINOR_VERSION);
+export const patch = Number(process.env.MUI_PATCH_VERSION);
+export const preReleaseLabel = process.env.MUI_PRERELEASE_LABEL || null;
+export const preReleaseNumber = Number(process.env.MUI_PRERELEASE_NUMBER) || null;
```

Comment:
I wonder if we shouldn't only expose the version, less bundle size.

```suggestion
```

e.g. React https://github.com/facebook/react/blob/main/packages/shared/ReactVersion.js

<img width="718" alt="SCR-20240826-ujde" src="https://github.com/user-attachments/assets/b200d8b0-4c8e-4a06-b75c-f49e71ba65b3">

or jQuery, I'm not aware they export more.

<img width="732" alt="SCR-20240826-uibg" src="https://github.com/user-attachments/assets/bf71c6d6-acf0-457e-9dd6-9bed4ac5d2f2">

Reply:
I get the value of version, major, minor, patch but I don't understand preReleaseLabel, preReleaseNumber.

I do believe that we should care it, bundle size is a fight of ever single instant: https://unpkg.com/browse/@mui/material@6.0.0/version/index.js shows:

```jsx
export const version = "6.0.0";
export const major = Number("6");
export const minor = Number("0");
export const patch = Number("0");
export const preReleaseLabel = undefined || null;
export const preReleaseNumber = Number(undefined) || null;
export default version;
```

I love Elon's point in https://www.youtube.com/watch?v=tRsxLLghL1k&t=574s: unless you have to add back 10% of what you remove, you are not removing enough.

Reply:
Bundle-size-wise I don't really mind either way. I expect these to have 0% impact on end-user bundles. These will be fully tree-shaken by minifiers if unused. 
But I agree  it's not clean (`isNaN(Number(undefined))`...). In hindsight I'd say just a `prerelease: string | undefined` export makes more sense to me.

Reply:
> bundle size is a fight of every single PR, if done systematically

Yes, this makes sense 👍🏼 

> I expect these to have 0% impact on end-user bundles. These will be fully tree-shaken by minifiers if unused.

That was my understanding as well.

So what option makes the most sense to you @oliviertassinari @Janpot?:

1. Replace `preReleaseLabel` and `preReleaseNumber` with `preRelease: string | undefined`
2. Remove `preReleaseLabel` and `preReleaseNumber`

Both could be considered breaking changes, but "not really" as those were only available in pre-release versions. If we want to be extra conservative, we can deprecate and remove them from the next major.

Reply:
From an API point of view changing it `prerelease: string | undefined` makes the most sense to me. But I don't have a strong opinion, I only see use for the major so far.
From a semver point of view it depends on how pedantic we want to be about this being a breaking change, but I think we can still kill it.

> though one could argue that if nobody imports them then we don't need them.

In the end we decided we'd just don't support css vars themes in toolpad when you're on v5 (which we initially did support), so we don't need this anymore at the moment. It keeps the door open for feature detection in the future. I really have no strong opinion on this one.

Reply:
I can see the purpose in having this long term: third party libraries who want to implement backward support between our different majors might appreciate that we have it. We could have it in, and see if this is used long term. 

I guess, I'm a bit wired like this 😁: https://youtu.be/tRsxLLghL1k?si=sRhbVRHVLRR83pMX&t=574. "if you are not adding back at least 10% of what you are removing, you know you are not removing enough".

Reply:
Well essentially the 10% that was removed was the `CssVarsProvider` as a separate theme provider. On toolpad side we couldn't know whether the `ThemeProvider` from our peer dependency supported css vars themes or not. (In 5 it doesn't and we wanted to fall back to `CssVarsProvider` in that case, as we were already doing). In the end we decided to break backwards compatibilty so the problem went away altogether. but at that time this was already added to @mui/material.

Essentially we took 10% out and added 0.1% back in. It's even better 😄